monitor.cpp

```diff
@@ -1,38 +1,42 @@
 #include "./monitor.h"
-#include <assert.h>
 #include <thread>
 #include <chrono>
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
+constexpr float MAX_TEMPERATURE = 102;
+constexpr float MIN_TEMPERATURE = 95;
+constexpr float MAX_PULSE_RATE = 100;
+constexpr float MIN_PULSE_RATE = 60;
+constexpr float MIN_SPO2 = 90;
+constexpr float MAX_SPO2 = 100;
+
+void printWarningGraphics() {
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+void printWarning(const char* vitalName) {
+    cout << vitalName << " is out of range!\n";
+    printWarningGraphics();
+}
+
+bool checkVitals(const char* vitalName, float value, float min, float max) {
+    if (value < min || value > max) {
+        printWarning(vitalName);
+        return false;
     }
-    return 0;
-  }
-  return 1;
+    return true;
+}
+
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    bool tempStatus = checkVitals("Temperature", temperature, MIN_TEMPERATURE, MAX_TEMPERATURE);
```

Comment:
Try naming the function, so that we can be clear about its purpose by reading just this line of code, without having to refer to the implementation.
Here, I get doubts like- `checkVitals` returns `true` to say that some attention is required? Or does `true` mean that everything is ok, no attention required?
Try a different name?